open_seq2seq/optimizers/optimizers.py

```diff
@@ -376,45 +375,35 @@ class should be sub-class of `tf.Optimizer` that implements
       )
 
     # LARC gradient re-scaling
-    if LARC_nu is not None and isinstance(LARC_nu, float):
+    if larc_params is not None:
+      check_params(
+        config=larc_params,
+        required_dict={'larc_nu': float},
+        optional_dict={
+          'larc_mode': ['clip', 'scale'],
+          'min_update': float,
+          'epsilon': float
+        },
+      )
+      larc_nu = larc_params['larc_nu']
+      larc_mode = larc_params.get('larc_mode', 'clip')
+      min_update = larc_params.get('min_update', 1e-7)
+      eps = larc_params.get('epsilon', 1e-7)
+
       for idx, (g, v) in enumerate(gradients):
         var_dtype = v.dtype
         v_norm = tf.norm(tensor=tf.cast(v, tf.float32), ord=2)
         g_norm = tf.norm(tensor=tf.cast(g, tf.float32), ord=2)
 
-        zero_const = tf.constant(0.0, dtype=tf.float32)
-        # this condition is necessary for two reasons.
-        # First, g_norm can be zero. In that case we can not use the usual
-        # formula, but it does not matter what larc_local_lr is, since it will
-        # be multiplied by g = 0.
-        # Second, v_norm can be zero. In that case, we want to switch to the
-        # usual gradient descent since the usual LARC update will always be 0.
-        # Thus, we make larc_local_lr = 1e-5 so that
```

Comment:
Lets also make sure that we have an option to track both updated and "raw" gradients in TB

Reply:
Added larc_summaries parameter with latest commits